### PR TITLE
Enhance Function component

### PR DESCRIPTION
**Pull request by xepozz**
- [X] Added ability to work with short closures
- [X] Added pointer cursor
- [X] Added tests
- [X] Added showcases

### CODE REVIEW

File: src/components/DataTypes/Function.tsx [hunk 1]
```diff
@@ -1,46 +1,62 @@
-import { Box } from '@mui/material'
+import {Box} from '@mui/material'
 import type React from 'react'
 
-import { useJsonViewerStore } from '../../stores/JsonViewerStore'
-import type { DataItemProps } from '../../type'
-import { DataTypeLabel } from '../DataTypeLabel'
+import {useJsonViewerStore} from '../../stores/JsonViewerStore'
+import type {DataItemProps} from '../../type'
+import {DataTypeLabel} from '../DataTypeLabel'
```
Comment (Himself65) [inline src/components/DataTypes/Function.tsx:6]: lint fix?

Comment (Himself65) [reply src/components/DataTypes/Function.tsx:6]: It should have a space

**Comment by Himself65** [comment]
Fixes: https://github.com/TexteaInc/json-viewer/issues/72

**Comment by Himself65** [comment]
Screenshot from 2022-09-23 08-29-27

still have the error

**Comment by Himself65** [comment]
There still have Hydration errors. let me check later

**Comment by Himself65** [comment]
> There still have Hydration errors. let me check later

I think that's because the server-side treat every function as the same(whether arrow function). but the client side is not.

Maybe we should turn NoSsr here?

**Comment by xepozz** [comment]
Yeap code is obfuscating by SSR and it should be disabled.
But the bug the function name doesn't match with it. I've fixed it.

**Comment by xepozz** [comment]
It's completely ready for review and merge.

**Review by Himself65 — APPROVED** [review]

**Pull request merged**